android/java/org/chromium/chrome/browser/crypto_wallet/util/CustomURLSpan.java

```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.crypto_wallet.util;
+
+import android.os.Parcel;
+import android.text.style.URLSpan;
+import android.text.TextPaint;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+
+public class CustomURLSpan extends URLSpan {
+
+    private View.OnClickListener mClickListener;
+    public CustomURLSpan(String url, View.OnClickListener clickListener) {
+        super(url);
+        mClickListener = clickListener;
+    }
+    public CustomURLSpan(String url) {
+        super(url);
+    }
+
+    public CustomURLSpan(@NonNull Parcel src) {
+        super(src);
+    }
+
+    @Override
+    public void onClick(View widget) {
+        mClickListener.onClick(widget);
+    }
+    
+    @Override
+    public void updateDrawState(@NonNull TextPaint ds) {
+        super.updateDrawState(ds);
+        ds.setUnderlineText(false);
+    }
+}
```

Comment:
no need of this, use [NoUnderlineClickableSpan](https://github.com/brave/brave-core/blob/e78b2cc3d743642111f7b544ebc73e4d656521c6/android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java#L856)